File: framework/TextBox.js
const BaseElement = require('./BaseElement.js')
const logger = require('./logger.js')

class TextBox extends BaseElement {
  constructor(selector, name) {
    super(selector, name)
  }

  async sendText(text) {
    logger.debug(`Sending text to ${this.name}`)   
    await (await this._getElement()).setValue(text)
    logger.debug(`sent text to ${this.name}: ${text}.`)    
  }

  async clear() {
    logger.debug(`Clearing ${this.name}`)    
    await (await this._getElement()).clearValue()
    logger.debug(`${this.name} cleared.`)    
  }
}

module.exports = TextBox

File: src/test/java/framework/elements/TextBox.java
package framework.elements;

import framework.utils.LoggerUtil;
import org.openqa.selenium.*;

public class TextBox extends BaseElement {
    public TextBox(By locator, String name){
        super(locator, name);
    }

    public void SendText(String text){
        LoggerUtil.debug("Sending text \"{}\" to element: {}", text, name);
        _GetElement().setValue(text);
    }

    public void CLear(){
        LoggerUtil.debug("Clearing text field: {}", name);
        _GetElement().clear();
    }

    public String GetPlaceholder(){
        var placeholder =  GetAttribute("placeholder");
        return placeholder;
    }
}
